src/rules/camelcase.rs

```diff
@@ -36,6 +36,36 @@ impl LintRule for Camelcase {
     visitor.visit_program(program, program);
     visitor.report_errors();
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Enforces the use of camelCase in variable names
+
+Consistency in a code base is key for readability and maintainability.  This rule
+enforces variable definitions and assignments to be in camelCase.  Of note:
+* `_` is allowed at the start or end of a variable
+* All uppercase variable names (e.g. constants) may have `_` in their name
+* Function calls are exempt
+* This rule also applies to variables imported via ES6 module imports
```

Comment:
It looks like this doesn't fully describe the `camelcase` rule to me.
It's a little hard to give a accurate description, but if I do my best to do it, it would be like...

```suggestion
Consistency in a code base is key for readability and maintainability.  This rule
enforces variable declarations and object property names to be in camelCase if they
are controllable on your own.  Of note:
* `_` is allowed at the start or end of a variable
* All uppercase variable names (e.g. constants) may have `_` in their name
* This rule also applies to variables imported / exported via ES module
```

Reply:
Thanks for the suggestion.  I think I need more help understanding this rule, as you say, it's really complex!  What do you mean by "if they are controllable on your own"?  Also, was `Function calls are exempt` removed because it was incorrect, too minor a detail, or too simplistic/badly worded?

Reply:
yeah it's really complex. 
As for "if they are controllable on your own", if I try to put it another way, it would be "if you can rename the variable name to camelCase at that point". 
Consider function call like `do_something();`, in this case we definitely cannot rename `do_something` to camelcase at this point. But say declaring function expression like `const f = function do_something() { };`, here we can rename `do_something` to `doSomething`. This is exactly what I want to describe.

So, `Function calls are exempt` is definitely correct, but it sounds a little too specific. The core of `camelcase` rule is that it only checks declarations, and this also covers the fact that function calls won't be checked.